engine/render/src/render/render.cpp

```diff
@@ -776,6 +849,9 @@ namespace dmRender
                     dmGraphics::DisableTexture(context, i, texture);
             }
         }
+
+        SetRenderStateIfChanged(context, ps_orig, dmGraphics::GetPipelineState(context));
```

Comment:
Isn't this the same issue as before?

Depending on what component type ends up rendered last, the state will be different for all components in the next call to Draw() ?

Reply:
Hm no I don't think so? we reset the render state to whatever was set before we entered this function (ps_orig)

Reply:
Gotcha.
I was confused by the name and functionality of the function.

Just like the other "Set" functions, I assumed it would set the _new_ value if it was different from the previous one.
Instead, it reverses the behavior, and it is like you commented, a "Reset" function.
Perhaps a comment and/or a function rename is in order?